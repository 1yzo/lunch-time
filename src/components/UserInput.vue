<template>
    <div id="input-container">
        <label for="user-input">What's your name?</label>
        <input
            id="user-input"
            type="text"
            :value="value" 
            autofocus
            @input="$emit('input', $event.target.value)"
            @keyup.enter="$emit('submit')"
        />
    </div>
</template>

<script>
export default {
    props: {
        value: String,
    }
}
</script>

<style lang="scss" scoped>
    label {
        font-size: 30px;
    }
    
    input {
        width: 100%;
        font-size: 30px;
        padding: 10px;
        border: none;
        box-shadow: inset 0 1px 5px  rgba(0, 0, 0, 0.199);
        margin-top: 10px;
        color: rgba(54, 69, 237, 0.637);
    }
</style>


