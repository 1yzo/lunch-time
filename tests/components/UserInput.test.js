import Vue from 'vue';
import { shallowMount } from '@vue/test-utils';
import UserInput from '../../src/components/UserInput';

describe('UserInput', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount(UserInput);
        expect(wrapper.element).toMatchSnapshot();
    });
});