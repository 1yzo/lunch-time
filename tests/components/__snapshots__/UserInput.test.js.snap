// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInput renders correctly 1`] = `
<div
  id="input-container"
>
  <label
    for="user-input"
  >
    What's your name?
  </label>
   
  <input
    autofocus="autofocus"
    id="user-input"
    type="text"
  />
</div>
`;
